docs(plugins/mtls-auth) Configuring mtls-auth-credentials with declarative config (#2365)
Kong/docs.konghq.com: app/_hub/kong-inc/mtls-auth/1.3-x.md

### Create manual mappings between certificate and Consumer object

> Sometimes, you may not want to use automatic Consumer lookup, or you may have certificates

Reviewer: ```suggestion
Sometimes, you might not want to use automatic Consumer lookup, or you have certificates
```

that contain a field value not directly associated with **Consumer** objects. In those
situations, you may manually assign one or more subject names to the **Consumer** object for
identifying the correct Consumer.

> > **Note:** "Subject names" refers to the certificate's Subject Alternative Names (SAN) or

Reviewer: I think italic font should be used very sparingly. It is more difficult for the reader's eye to scan.

Author: This is the standard blockquote styling in Markdown - we generally use it interchangeably with the blue info blocks (`div class="alert alert-ee blue"`). Maybe we should decide, as a team, which info blocks to use for notes? And then I can change the CSS for note formatting to match.

Author: Leaving as is for now. Will revisit note styling with team.

}
```

{% endnavtab %}
{% navtab Declarative (YAML) %}

To create a subject name mapping using declarative configuration, you will need
to generate a UUID for each `mtls_auth_credentials` mapping. You can use any
UUID generator to do this. Here are some common ones, depending on your OS:
* [Linux](https://man7.org/linux/man-pages/man1/uuidgen.1.html)
* [MacOS](https://brunty.me/post/uuidgen-on-macos/)
* [Windows](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/new-guid?view=powershell-7)

> Once you have a UUID, add the following to your declarative configuration

Reviewer: ```suggestion
After you have generated a UUID, add the following to your declarative configuration
```

file:

```yaml
consumers:
- custom_id: my-consumer
  username: {consumer}
  mtls_auth_credentials:
  - id: bda09448-3b10-4da7-a83b-2a8ba6021f0c
    subject_name: test@example.com
```

{% endnavtab %}
{% endnavtabs %}

#### Parameters for manual mapping

Form Parameter                            | Default | Description
---                                       | ---     | ---
`id`<br>*required for declarative config* |  none   | UUID of the Consumer-mapping. Required if adding mapping using declarative configuration, otherwise generated automatically by Kong's Admin API.
`subject_name`<br>*required*              |  none   | The Subject Alternative Name (SAN) or Common Name (CN) that should be mapped to `consumer` (in order of lookup).

> `ca_certificate`<br>*optional*            |  none   | **If using the Kong Admin API:** UUID of the Certificate Authority (CA). <br><br> **If using declarative configuration:** Full PEM-encoded CA certificate. <br><br>The provided CA UUID or full certificate has to be verifiable by the issuing certificate authority for the mapping to succeed. This is to help distinguish multiple certificates with the same subject name but are issued under different CAs. <br><br>If empty, the subject name will match certificates issued by any CA under the corresponding `config.ca_certificates`.

Reviewer: ```suggestion
`ca_certificate`<br>*optional*            |  none   | **If using the Kong Admin API:** UUID of the Certificate Authority (CA). <br><br> **If using declarative configuration:** Full PEM-encoded CA certificate. <br><br>The provided CA UUID or full certificate has to be verifiable by the issuing certificate authority for the mapping to succeed. This is to help distinguish multiple certificates with the same subject name that are issued under different CAs. <br><br>If empty, the subject name will match certificates issued by any CA under the corresponding `config.ca_certificates`.
```

### Matching behaviors

> Once a client certificate has been verified as valid, the **Consumer** object will be determined in the following order, unless `skip_consumer_lookup` is set to `true`:

Reviewer: This could happen more than one time. Avoid future/passive tense.

```suggestion
After a client certificate has been verified as valid, the **Consumer** object is determined in the following order, unless `skip_consumer_lookup` is set to `true`:
```

1. Manual mappings with `subject_name` matching the certificate's SAN or CN (in that order) and `ca_certificate = <issuing authority of the client certificate>`
2. Manual mappings with `subject_name` matching the certificate's SAN or CN (in that order) and `ca_certificate = NULL`
3. If `config.consumer_by` is not null, Consumer with `username` and/or `id` matching the certificate's SAN or CN (in that order)
4. The `config.anonymous` consumer (if set)

> > **Note**: Matching will stop as soon as the first successful match is found.

Reviewer: ```suggestion
**Note**: Matching stops as soon as the first successful match is found.
```

### Troubleshooting

> When authentication fails, the client does not have access to any details explaining the

Reviewer: // tense

```suggestion
When authentication fails, the client does not have access to any details that explain the
```
